client/src/index.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class App extends React.Component {
+  //because I'm not using state, with more time I would probably restyle this to a functional component
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick() {
+    let sheets = document.styleSheets;
+    let sheet;
+    let newCSS = document.querySelector('#css').value;
+    if (sheets.length === 0) {

```

Comment:
What I did was create a sheet when the app loads, then just update its contents on user action. Will save you the checks and the `if`s.

```
const sheet = document.createElement("style");
sheet.innerHTML = code;
document.body.appendChild(sheet);
```

```
function onChange(newValue) {
    setStyle(newValue);
    sheet.innerHTML = newValue;
}
```